### ISSUE TITLE
failed to create docker client, empty docker host

### ISSUE THREAD

**Issue by toxik-io** [containous/traefik#5023]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

defined a docker endpoint, which traefik interpreted as a blank string
### What did you expect to see?

Traefik connecting to the docker endpoint.

### What did you see instead?

Traefik showing a blank string for the host, in the error message.

related to these function calls:

* ParseHostUrl V1.7 https://github.com/containous/traefik/blob/v1.7/provider/docker/docker.go#L97
* CreateClient V1.7 https://github.com/containous/traefik/blob/v1.7/provider/docker/docker.go#L132
* ParseHostUrl Master https://github.com/containous/traefik/blob/890d02638b8673ecc593764966cec794a1b3e930/pkg/provider/docker/docker.go#L123
* CreateClient Master https://github.com/containous/traefik/blob/890d02638b8673ecc593764966cec794a1b3e930/pkg/provider/docker/docker.go#L161

no idea why p.endpoint is empty... because I don't have a traceback on this...

...more below

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.12
Codename:     maroilles
Go version:   go1.11.10
Built:        2019-05-29_07:35:02PM
OS/Arch:      linux
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
[provider.docker]
#endpoint = "unix:///var/run/docker.sock" # using this doesn't work (dunno why)
endpoint = "tcp://8.8.8.8:2375" # also tried the domain... both are blank in the error...
domain = "example.com" # coincidentally same as the global domain
watch = true
exposedbydefault = true
usebindportip = true
#swarmmode = true # tried both ways at every step...


[docker.tls] # not related... hopefully
PEM = true
CAOptional = true
ca = "/etc/ssl/ca.pem" # CA is specific to the domain, not an IP
cert = "/etc/ssl/cert.pem"
key = "/etc/ssl/key.pem"
insecureskipverify = true

```
generated certs used in a terraform docker swarm cluster (which works on the standard docker client, had a few CA issues in traefik and had to set CAOptional to true...)... also it expects the cert/key to be of the client, not cross-verifying the cert of the server... which is weird... expecting them to match and all... anyway... the CA is the root of trust for the server key regardless... I'm off topic...
```dockerfile
FROM traefik:latest
COPY traefik.toml /traefik.toml
COPY infra/.docker/ca.pem /etc/ssl/ca.pem
COPY infra/.docker/cert.pem /etc/ssl/cert.pem
COPY infra/.docker/key.pem /etc/ssl/key.pem
```

The important bit for the compose file in this issue is that I've tried to add the DOCKER_HOST environment variable to see if that helps... it did nothing:

```yaml
version: "3.7" # for the volumes bind option.
services:
  traefik:
    image: traefik:latest # docker stack likes reference images, and is fooled by this.
    build: # the actual image used
      context: .
      dockerfile: traefik.dockerfile
    environment:
      - 'CLOUDFLARE_EMAIL="${cloudflareemail}"'
      - 'CLOUDFLARE_API_KEY="${cloudflaretoken}"'
      - 'constraint:node==manager'
      - 'DOCKER_HOST=tcp://8.8.8.8:2375' # for good measure... still blank.
    # expose:
    #   - '80'
    #   - '443'
    #   - '8083'
    labels:
      - traefik.enable=true
      - traefik.backend=traefik
      - traefik.backend.loadbalancer.method=wrr
      # - traefik.docker.network=fo_front-tier
      - 'traefik.frontend.rule=Host:traefik.${fo_host}'
      - traefik.frontend.passHostHeader=true
      - traefik.port=7000
      - traefik.protocol=http
    # networks:
    #   - front-tier
    #   - back-tier
    ports:
      - '80:80'
      - '443:443'
      - '8080:8080'
    restart: always
    deploy:
      mode: replicated
      labels:
        - traefik.enable=true
        - traefik.backend=traefik
        - traefik.backend.loadbalancer.method=wrr
        # - traefik.docker.network=fo_front-tier
        - 'traefik.frontend.rule=Host:traefik.${fo_host}'
        - traefik.frontend.passHostHeader=true
        - traefik.port=8080
        - traefik.protocol=http
      placement:
        constraints: # for later swarm business...
          - node.role == manager
    volumes:
#      - /etc/ssl:/etc/ssl # using a dockerfile instead
      - type: bind # because the docker socket wasn't initially working
        source: /var/run/docker.sock # but I'm off topic.
        target: /var/run/docker.sock
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
traefik_1     | time="2019-06-25T17:45:49Z" level=error msg="Provider connection error unable to parse docker host ``, retrying in 719.790051ms"
traefik_1     | time="2019-06-25T17:45:50Z" level=error msg="Failed to create a client for docker, error: unable to parse docker host ``"
traefik_1     | time="2019-06-25T17:45:50Z" level=error msg="Provider connection error unable to parse docker host ``, retrying in 602.173973ms"
traefik_1     | time="2019-06-25T17:45:50Z" level=error msg="Failed to create a client for docker, error: unable to parse docker host ``"
traefik_1     | time="2019-06-25T17:45:50Z" level=error msg="Provider connection error unable to parse docker host ``, retrying in 1.455304865s"
```

## In Conclusion
Do we have steps I can take on my host to give you better debug info for why the endpoint is blank on that line of code?

**Comment by dtomcej** [comment]
`[provider.docker]` is the provider configuration for Version 2.

`[docker]` is the provider configuration for version 1.

Try:
```
[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "example.com"
watch = true
exposedbydefault = true
swarmmode = true
```

**Comment by toxik-io** [comment]
I mean, maybe when it reads the SANs it filters out domains and only uses IPs.. and then overwrites the docker_host variable... there's loads of ways this is odd. I could paste the SAN summary from cfssl if it's that...

**Comment by toxik-io** [comment]
oh, well, that's dumb.
